k8s/apps.go

```diff
@@ -549,24 +549,22 @@ func streamPodOutput(k *k8sHelper, pod *api.Pod, opts *api.PodLogOptions) (strea
 
 func (c apps) newPodListLogger(podList *api.PodList, opts *api.PodLogOptions) io.ReadCloser {
 	var wg sync.WaitGroup
-	wg.Add(len(podList.Items))
 	r, w := io.Pipe()
 	for _, pod := range podList.Items {
-		p := &pod
+		tmp := pod
+		if tmp.Status.Phase != api.PodRunning {
+			continue
+		}
+		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			rc, err := streamPodOutput(c.k, p, opts)
+			rc, err := streamPodOutput(c.k, &tmp, opts)
 			if err != nil {
 				log.Errorf("reading logs %s", err)
```

Comment:
I'd take the chance you're touching this code to improve this log message: `reading logs` doesn't help much.